tesseract_process_managers/include/tesseract_process_managers/core/task_info.h

```diff
@@ -52,7 +52,7 @@ class TaskInfo
   using Ptr = std::shared_ptr<TaskInfo>;
   using ConstPtr = std::shared_ptr<const TaskInfo>;
 
-  TaskInfo(std::size_t unique_id, std::string name = "");
+  TaskInfo(std::size_t unique_id = 0, std::string name = "");
```

Comment:
How is the unique_id normally previously set?

Reply:
It is set at the beginning of conditionalProcess for each TaskGenerator. It should still be set. This is just for serialization.

Reply:
Oh, yea. I forgot for serialization it must have a default constructor. Maybe add documentation state that default value is only provided for serialization developer should provide it?
 
or something like this. 

``` c++
TaskInfo() = default // Required for serialziation
TaskInfo(std::size_t unique_id, std::string name = "");
```

Reply:
Yeah that's probably better. I changed it to that.